### ISSUE TITLE
[Bug] Ryujinx version 1.1.1350 (Latest) crashes immediately on boot on M1 due to "ResultFsSha256PartitionHashVerificationFailed"

### ISSUE THREAD

**Issue by enbyte** [Ryujinx/Ryujinx#7045]
### Description of the issue

Ryujinx crashes immediately when being opened on Sonoma 14.5. When installing with a clean `Library/Application Support/Ryujinx`, it did not crash. I could add my prod keys, and reopening it did not crash. I then added my firmware, which did not crash, and loaded some NSPs from an external disk. I then applied updates to two of these NSPs. Every time opening after that, Ryujinx crashes right away and pops up the little "x quit unexpectedly" window. I reinstalled by just deleting the app but not clearing the Ryujinx library folder, which led to the same issue. 

https://github.com/Ryujinx/Ryujinx/issues/1805 mentioned that if there were no logs in `Contents/MacOS`, it was likely a Gatekeeper issue. However, on install, I went through the whole "blah blah apple says this is dangerous" garbage, and allowed it to run from Security/Privacy in the system settings. 

https://github.com/Ryujinx/Ryujinx/issues/4151 says it was a "permissions issue", and offers some command to reset user permissions for your drive. I ran this command for both my computer hard drive (where ryujinx + data is stored) and the external hard drive (where the game files are stored). When initially adding the games, a dialog popped up with "Allow Ryujinx to access files on a removable volume?" which I hit "OK" on. It loaded the games fine and I booted and played a minute of BOTW with no issues. Just to test, I also gave Ryujinx the "Full Disk Access" permission, which changed nothing. I ran it as root, which then created a new Ryujinx folder in `/var/root/Library/Application Support/Ryujinx/Config.json` and did not crash with the empty folder, same as it was when freshly installed and running with my home directory as the home directory.

Any ideas what could be causing this? My guess is some sort of input validation error or issue with loading the folder data, since with clean folder data it does not crash but does with keys, firmware, games, etc. Looking in the console when it is run looks like a SHA-256 verification error, which is strange since I have not changed anything on the disk or messed with the files. The games loaded fine when initially running the "Add games from a folder", and BOTW booted with no issues.

Console log:
```00:00:00.003 |N| Application ReloadConfig: Loading configuration from: /Users/[redacted]/Library/Application Support/Ryujinx/Config.json
00:00:00.115 |I| Configuration LogValueChange: EnableFileLog set to: True
00:00:00.149 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.154 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.154 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.154 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.155 |I| Configuration LogValueChange: BackendThreading set to: Auto
00:00:00.155 |I| Configuration LogValueChange: GraphicsBackend set to: Vulkan
00:00:00.155 |I| Configuration LogValueChange: PreferredGpu set to: 0x106B_0xE0503EF
00:00:00.156 |I| Configuration LogValueChange: AntiAliasing set to: None
00:00:00.156 |I| Configuration LogValueChange: ScalingFilter set to: Bilinear
00:00:00.156 |I| Configuration LogValueChange: ScalingFilterLevel set to: 80
00:00:00.157 |I| Configuration LogValueChange: EnableDockedMode set to: False
00:00:00.161 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.161 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.161 |I| Configuration LogValueChange: EnableTextureRecompression set to: False
00:00:00.161 |I| Configuration LogValueChange: EnableMacroHLE set to: True
00:00:00.161 |I| Configuration LogValueChange: EnableColorSpacePassthrough set to: False
00:00:00.161 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.161 |I| Configuration LogValueChange: EnableInternetAccess set to: False
00:00:00.161 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.161 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.161 |I| Configuration LogValueChange: AudioBackend set to: SDL2
00:00:00.161 |I| Configuration LogValueChange: AudioVolume set to: 1
00:00:00.162 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.162 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.162 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.162 |I| Configuration LogValueChange: UseHypervisor set to: True
00:00:00.163 |I| Configuration LogValueChange: MultiplayerMode set to: Disabled
00:00:00.164 |N| Application PrintSystemInfo: Ryujinx Version: 1.1.1350
00:00:00.166 |N| Application Print: Operating System: macOS 14.5.0 (23F79) (Arm64)
00:00:00.166 |N| Application Print: CPU: Apple M1 ; 8 logical
00:00:00.167 |N| Application Print: RAM: Total 16 GiB ; Available 7.4 GiB
00:00:00.168 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
00:00:00.169 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:01.361 |E| GUI.ApplicationLibraryThread Application : Unhandled exception caught: LibHac.Common.HorizonResultException: ResultFsSha256PartitionHashVerificationFailed (2002-4643)
   at LibHac.Common.ThrowHelper.ThrowResult(Result result)
   at LibHac.Result.ThrowIfFailure()
   at LibHac.Tools.FsSystem.NcaUtils.NcaHeader.DecryptHeader(KeySet keySet, IStorage storage)
   at LibHac.Tools.FsSystem.NcaUtils.NcaHeader..ctor(KeySet keySet, IStorage headerStorage)
   at LibHac.Tools.FsSystem.NcaUtils.Nca..ctor(KeySet keySet, IStorage storage)
   at Ryujinx.HLE.Loaders.Processes.Extensions.PartitionFileSystemExtensions.GetNca(IFileSystem fileSystem, KeySet keySet, String path) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.HLE/Loaders/Processes/Extensions/PartitionFileSystemExtensions.cs:line 159
   at Ryujinx.HLE.Loaders.Processes.Extensions.PartitionFileSystemExtensions.GetContentData(IFileSystem partitionFileSystem, ContentMetaType contentType, VirtualFileSystem fileSystem, IntegrityCheckLevel checkLevel) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.HLE/Loaders/Processes/Extensions/PartitionFileSystemExtensions.cs:line 35
   at Ryujinx.UI.App.Common.ApplicationLibrary.GetApplicationsFromPfs(IFileSystem pfs, String filePath) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 178
   at Ryujinx.UI.App.Common.ApplicationLibrary.TryGetApplicationsFromFile(String applicationPath, List`1& applications) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 290
   at Ryujinx.UI.App.Common.ApplicationLibrary.LoadApplications(List`1 appDirs, Language desiredTitleLanguage) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 555
   at Ryujinx.Ava.UI.Windows.MainWindow.<ReloadGameList>b__77_0() in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx/UI/Windows/MainWindow.axaml.cs:line 625
   at System.Threading.Thread.StartCallback()
Unhandled exception. LibHac.Common.HorizonResultException: ResultFsSha256PartitionHashVerificationFailed (2002-4643)
   at LibHac.Common.ThrowHelper.ThrowResult(Result result)
   at LibHac.Result.ThrowIfFailure()
   at LibHac.Tools.FsSystem.NcaUtils.NcaHeader.DecryptHeader(KeySet keySet, IStorage storage)
   at LibHac.Tools.FsSystem.NcaUtils.NcaHeader..ctor(KeySet keySet, IStorage headerStorage)
   at LibHac.Tools.FsSystem.NcaUtils.Nca..ctor(KeySet keySet, IStorage storage)
   at Ryujinx.HLE.Loaders.Processes.Extensions.PartitionFileSystemExtensions.GetNca(IFileSystem fileSystem, KeySet keySet, String path) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.HLE/Loaders/Processes/Extensions/PartitionFileSystemExtensions.cs:line 159
   at Ryujinx.HLE.Loaders.Processes.Extensions.PartitionFileSystemExtensions.GetContentData(IFileSystem partitionFileSystem, ContentMetaType contentType, VirtualFileSystem fileSystem, IntegrityCheckLevel checkLevel) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.HLE/Loaders/Processes/Extensions/PartitionFileSystemExtensions.cs:line 35
   at Ryujinx.UI.App.Common.ApplicationLibrary.GetApplicationsFromPfs(IFileSystem pfs, String filePath) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 178
   at Ryujinx.UI.App.Common.ApplicationLibrary.TryGetApplicationsFromFile(String applicationPath, List`1& applications) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 290
   at Ryujinx.UI.App.Common.ApplicationLibrary.LoadApplications(List`1 appDirs, Language desiredTitleLanguage) in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx.UI.Common/App/ApplicationLibrary.cs:line 555
   at Ryujinx.Ava.UI.Windows.MainWindow.<ReloadGameList>b__77_0() in /home/runner/work/Ryujinx/Ryujinx/src/Ryujinx/UI/Windows/MainWindow.axaml.cs:line 625
   at System.Threading.Thread.StartCallback()
zsh: abort      ./Ryujinx
```

There are no log files to attach, but here is the console crash report:
https://pastebin.com/0VaMjPGX (Pasting it here made github complain that the issue was too long)

### Reproduction steps

Install latest version of Ryujinx on Sonoma 14.5 on M1. Add firmware and keys. Then, load games from an external drive. Quit Ryujinx and reopen it.

### Log file

N/A

### OS

MacOS Sonoma 14.5

### Ryujinx version

1.1.1350

### Game version

No games being tested, Ryujinx will not start

### CPU

M1

### GPU

M1 internal GPU

### RAM

16GB

### List of applied mods

N/A

**Comment by piplup55** [comment]
your game file is corrupted, you need to redump it

**Comment by enbyte** [comment]
Is it possible for Ryujinx to skip the file and boot or at least output which one is corrupted instead of crashing?

**Comment by gdkchan** [comment]
I think its supposed to be fixed by #7036.

**Comment by enbyte** [comment]
The artifact from the mentioned PR boots without crashing. Seems to be fixed.
